tower-http/src/auth/require_authorization.rs

```diff
@@ -1,3 +1,7 @@
+#![deprecated(
+    since = "0.6.6",
+    note = "uses non-constant time string comparison for tokens which could allow timing attacks"
```

Comment:
Tbh, not a fan of this deprecation note. If this was the only problem with this middleware, I don't think that would be a reason to deprecate it. Rather, it's

- as Sean noted, this middleware just isn't applicable to real-world scenarios due to how limited it is
- we don't have the resources / motivation to turn it into something useful
- writing your own without this is very easy, especially with something like axum's `middleware::from_fn`